src/UICircularProgressRing/UICircularProgressRing.swift

```diff
@@ -1045,14 +1048,31 @@ fileprivate extension CALayer {
         ringLayer.animationDuration = duration
 
         CATransaction.begin()
-        CATransaction.setCompletionBlock {
-            self.delegate?.didFinishProgress?(for: self)
-            completion?()
+        
+        //Store the completion event locally
+        self.completion = completion
+        
+        //Check if a completion timer is still active and if so stop it
+        if let completionTimer = completionTimer {
+            completionTimer.invalidate()
+            self.completionTimer = nil
         }
+        
+        //Create a new completion timer
+        completionTimer = Timer.scheduledTimer(timeInterval: duration,
```

Comment:
can you use the `scheduledTimer(withTimeInterval:repeats:block:)` function instead, so you don't require a separate selector

Reply:
I think that one wasn't supported until iOS 10.0+, I'll give it another look though, might have been a slightly different scheduledTimer call